astropy/timeseries/periodograms/lombscargle/implementations/chi2_impl.py

```diff
@@ -78,6 +78,9 @@ def compute_power(f):
         XTy = np.dot(X.T, yw)
         return np.dot(XTy.T, np.linalg.solve(XTX, XTy))
 
+    p = np.array([compute_power(f) for f in frequency])
+    p = np.array(chi2_ref - [compute_power(f) for f in frequency])
```

Comment:
these look like they were committed accidentally ?

Reply:
They were not, but the change to lstsq solver made the fastchi2 method fail the tests, so I've undone it, as it's mostly a cosmetic change. Generally the dot product in chi² is mathematically equivalent to the chi2_ref - residuals.

Generally lstsq (how I've replaced it before undoing the changes) *should* be more stable, but it resulted in the fast chi² method failing all of the axcuracy tests for near-zero results (with errors ≤ 10^-12) so the only alternative to pass the tests was loosening the toleances.

I'm still waiting for feedback from someone else, generally I would prefer to reapply theese changes, but changing the tests without wider community's feedback only because NumPy's recommended function for given purpose fails them doesn't sound like a good idea, so I've removed suggested changes, at least for now.

Reply:
just noting that I checked, and reverting these lines doesn't cause any test to fail